as/cNFT/assembly/models/persistent_market.ts

```diff
@@ -168,9 +168,15 @@ export class PersistentMarket {
         tokenId: TokenId,
         accountId: AccountId
     ): PersistentSet<TokenId> {
-        let _set = this._amap.getSome(accountId)
-        _set.delete(tokenId)
-        return _set
+        let _set = this._amap.get(accountId)
+        if(!_set){
+            _set = new PersistentSet<TokenId>('_abs' + accountId)
```

Comment:
Why would you create a new PersistentSet if you are trying to remove a `TokenId`? If `_set` does not exist it means that the user has no bids on any tokens and this set will be created in case he adds a `Bid` to a token.
I guess the reason this is here is cause function is supposed to return `PersistenSet<TokenId>` but in general it's better to move this sort of checks higher up in the call stack.

Reply:
I would check for this in the `remove` method above.

Reply:
Yeah. you're right. I just checked this with the new contract. I didn't get this error this time. Seems like it has solved with this. https://github.com/curaOS/contracts/pull/100

Before that PR, me and @achrafismyname tested about this error. This was the only way that It solved for us. But I think now we can safely ignored this PR and close it. Thanks for pointing it out.